src/Api/Client.php

```diff
@@ -115,15 +120,19 @@ public function delete(string $path): ResponseInterface
      *
      * @param string $routeName
      * @return ResponseInterface
+     *
+     * @internal
      */
-    protected function send(string $method, string $path): ResponseInterface
+    public function send(string $method, string $path): ResponseInterface
     {
         $request = ServerRequestFactory::fromGlobals(null, $this->queryParams, $this->body)
+            ->withParsedBody($this->body)
```

Comment:
Is `$this->body` really the same variable for both the original body and the parsed body? Wouldn't those two values be different?

Reply:
https://github.com/flarum/core/blob/c1992b45debaacae6024d36d1b32a32fb129b312/src/Http/Middleware/ParseJsonBody.php#L18-L18 converts body to json, but the provided body must already be an array. With this change, any code expecting an array as parsedBody will see exactly that.

Perhaps we should json encode the original body?

Reply:
I wasn't sure when I wrote the comment but now I see that the ParseJsonBody middleware is part of the excluded middlewares.

Can't we copy both the parsed and non-parsed body from the parent request?

JSON encoding sounds like a good alternative, but it requires checking if the body has actually been parsed, because if it's not JSON in the first place this will mess things up.

I suppose it's rare for an extension to read the original raw body WHILE it's a valid JSON, but I feel like this should be consistent to avoid any issue.

Reply:
> Can't we copy both the parsed and non-parsed body from the parent request?

Almost always, the body being sent in an API client request will not be the same as the body from the original request. As opposed to the actor and session, which should be kept.

Reply:
Actually, this turned out to be unnecessary, as ::fromGlobals sets the provided body as parsed body, not as raw body.